### PR TITLE
tldr: init at 1.0

### PR DESCRIPTION
**Pull request by Radvendii**
Info on tldr: https://github.com/tldr-pages/tldr
The package added is for the C++ client.

### CODE REVIEW

**Comment by nckx** [comment]
(Previously: https://github.com/NixOS/nixpkgs/pull/12117)

**Comment by nckx** [comment]
Go ahead and squash both commits into one, since one won't work without the other.

**Comment by fpletz** [comment]
And could you please add `meta` attributes?

**Comment by Radvendii** [comment]
Yes. Should I squash that as well? (sorry, I'm new to this)

**Comment by nckx** [comment]
Yep.

And no reason to be sorry. New users == good.

**Comment by nckx** [comment]
New contributors == even better.

**Comment by Radvendii** [comment]
Thanks :). What fields of meta data should I fill in? All of them? I'm not sure about "priority" and "platforms"; I've only tested it on NixOS.

File: pkgs/tools/misc/tldr/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{stdenv, clang, fetchurl, curl}:
+
+stdenv.mkDerivation {
+  name = "tldr-1.0";
+  src = fetchurl {
+      url = "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.0.tar.gz";
+          sha256 = "11k2pc4vfhx9q3cfd1145sdwhis9g0zhw4qnrv7s7mqnslzrrkgw";
+            };
+  buildInputs = [curl clang];
```
Comment (nckx) [inline pkgs/tools/misc/tldr/default.nix]: Your indentation nesting is still off. Your text editor might need more convincing.

**Comment by Radvendii** [comment]
I *think* i fixed everything that we talked about.

**Comment by nckx** [comment]
`priority` isn't needed.

I tend to be very conservative about `platforms` and only set it to `linux` that I can test myself. For non-essential packages that's usually fine, those wanting other platforms can quickly test & add them.

**Comment by nckx** [comment]
[Almost there.](https://travis-ci.org/NixOS/nixpkgs/builds/100011255)

```
error: syntax error, unexpected '=', expecting ';', at /home/travis/build/NixOS/nixpkgs/pkgs/tools/misc/tldr/default.nix:10:21
```

**Comment by Radvendii** [comment]
Yep. Saw that. Fixed it. Also removed Mac from supported platforms.

**Comment by Radvendii** [comment]
Is version a string?

**Comment by nckx** [comment]
A string, since one can have versions like `1.3-rc7-rev16`.

Duplicating it is also a recipe for confusion later on, so don't do that (also in `src`). My preferred method: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/graphics/swingsane/default.nix.

**Comment by nckx** [comment]
```
nckx@dm1 ~/Hacking/nixpkgs master $ tldr ls

 ls
 List directory contents

 List all files, even hidden
   ls -a

...
```

Awesome! Thanks.

**Comment by Radvendii** [comment]
Wait what? That means it's done? You've added tldr to nixpkgs? What was all that about ls?

**Comment by nckx** [comment]
Just me testing it? And yes, it's been added.

**Comment by Radvendii** [comment]
Haha. Yeah, I realized after I asked what happened. Thanks so much for all the help :D

**Pull request merged**